lib/class-wp-theme-json-gutenberg.php

```diff
@@ -835,6 +837,10 @@ private function get_block_styles( $style_nodes, $setting_nodes ) {
 			$selector     = $metadata['selector'];
 			$declarations = self::compute_style_properties( $node );
 			$block_rules .= self::to_ruleset( $selector, $declarations );
+
+			if ( self::ROOT_BLOCK_SELECTOR === $selector ) {
+				$block_rules .= '.wp-site-blocks > * + * { margin-top: var( --wp--style--block-gap ); margin-bottom: 0; }';
```

Comment:
On theme.json, If I apply blockGap rule inside core/group block it seems we are not handling this case I guess we could handle it like we do on the layout "$selector > * + * { margin-top: var( --wp--style--block-gap ); margin-bottom: 0; }";?

Reply:
Not sure how to do that, would you mind helping here?

Reply:
Making as resolved because what we have already works.